tests/e2e.test.js

```diff
@@ -295,17 +296,34 @@ describe('Ordering', () => {
     }
   });
 
-  it.skip('orders grid items by color', async () => {
-    // Items are ordered using the CSS property `order`...
-    await expect(page).toClick('#order-color');
+  it('orders grid items by color', async () => {
+    const $items = await page.$$('li.grid-item');
+    for (let i = 0; i < $items.length; i++) {
+      const $button = await $items[i].$('button.grid-item__color');
 
-    const $gridItems = await page.$$('.copy-color');
-    for (let i = 0; i < $gridItems.length; i++) {
-      const $gridItem = $gridItems[i];
-      const hex = hexes[i];
-      await expect($gridItem).toMatch(`#${hex}`);
+      const $text = await $items[i].evaluate((el) =>
+        JSON.stringify(el.getAttribute('style')),
+      );
+      const $idx = parseInt($text.substring(15, $text.length - 1));
+
+      const $color = await $button.evaluate((el) =>
+        JSON.stringify(el.textContent),
+      );
+
+      await expect($color).toMatch(`#${hexes[$idx]}`);
     }
   });
+
+  // issue: https://github.com/simple-icons/simple-icons-website/issues/66
+  it('compares the first two colors to make sure they are not black', async () => {
+    // should not match black colors
+    await expect(hexes[0]).not.toMatch('181717');
+    await expect(hexes[1]).not.toMatch('1D1717');
+
+    // should match red colors
+    await expect(hexes[0]).toMatch('BE3939');
+    await expect(hexes[1]).toMatch('CB3837');
```

Comment:
I think these tests are way to specific and it's better to just remove it.

The first two asserts only verify these exact two colors don't appear there, not accounting for other colors. Even if these are the only two colors that can appear here, it assumes way too much about implementation in my opinion.

The last to asserts break as soon as there's a brand that has a color that should appear before "The Spriters Resource". Again, even if these must always be the first two colors (which is actually impossible, as there may be two brands that have `#BE3939`), it assumes way too much about implementation in my opinion.

Reply:
yes, that is correct. That's the way the test should be: very specific.
Perhaps it's too specific for end to end testing, and this would be better in an unit test suite. Do we have one of those?
The idea of the test case is to be specific, and to break easily. Perhaps the last two is too much detail, I would agree on that, but the first two are certainly important to make sure the issue in the comment is addressed.And they appear again, it's either because they are the lightest colors in the collection (highly unlikely) or there is an error, in which case the test should break.

In short: 
- The case is specific by design.
- The last two matches may be too specific, I'll be happy to remove them. The first two however I think are good where they are.
- This may be more suited for unit testing than e2e. Please advice if we have a more fitting place to put this testcase.

Reply:
I agree that sometimes specific tests are important, the reason I disagree in this case is that we're testing something very specific about something that is not in our control. This test can be broken by a non-breaking change in one of our dependencies!

Reply:
how? the only way to break this is to have those two colors close to black appearing at the beginning of the list (notice that I removed the last two assertions).

for that to happen it require changes to the color srting algo or removing thousands of icons.

also, this is independent of all dependencies, AFAIK. Which dependency affect this test?

Reply:
What if GitHub's color changes from `#181717` to `#191717`? Then `await expect(hexes[0]).not.toMatch('181717');` is pointless.

> notice that I removed the last two assertions

That is already better. My last comment was indeed mostly focused on the last two assertions.

Reply:
yes, if it changes the test will be pointless. However the idea of the unit test is to avoid regressions. If the color changes and the test case is found to be pointless, **then** it can be eliminated. That's the nature of unit tests, to evolve as conditions change.

Reply:
> If the color changes and the test case is found to be pointless, then it can be eliminated.

And how will you know, in the context of this repository, when it changes? If the `hexes` were part of the source code I would have no objections.

If you want to actually unit test this, then you should create a unit test suite for the script you added (`scripts/color-sorting.js `) and test it with a fixed list of hexes. As long as your test depends on the simple-icons dependency it is not a unit test, it is at least an integration test.

Reply:
I've moved the testing case to ordering.test.js and added a mock color array to make the input fixed.
Check it, plz.